libs/langchain-community/src/document_loaders/web/reddit.ts

```diff
@@ -0,0 +1,133 @@
+import { BaseDocumentLoader } from "@langchain/core/document_loaders/base";
+import { Document } from "@langchain/core/documents";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import {
+  RedditAPIWrapper,
+  RedditPost,
+  RedditAPIConfig,
+} from "../../utils/reddit.js";
+
+/**
+ * Class representing a document loader for loading Reddit posts. It extends
+ * the BaseDocumentLoader and implements the RedditAPIConfig interface.
+ * @example
+ * ```typescript
+ * const loader = new RedditPostsLoader({
+ *   clientId: "REDDIT_CLIENT_ID",
+ *   clientSecret: "REDDIT_CLIENT_SECRET",
+ *   userAgent: "REDDIT_USER_AGENT",
+ *   searchQueries: ["LangChain", "Langchaindev"],
+ *   mode: "subreddit",
+ *   categories: ["hot", "new"],
+ *   numberPosts: 5
+ * });
+ * const docs = await loader.load();
+ * ```
+ */
+export class RedditPostsLoader
+  extends BaseDocumentLoader
+  implements RedditAPIConfig
+{
+  public clientId: string;
+
+  public clientSecret: string;
+
+  public userAgent: string;
+
+  private redditApiWrapper: RedditAPIWrapper;
+
+  private searchQueries: string[];
+
+  private mode: string;
+
+  private categories: string[];
+
+  private numberPosts: number;
+
+  constructor({
+    clientId = getEnvironmentVariable("REDDIT_CLIENT_ID") as string,
+    clientSecret = getEnvironmentVariable("REDDIT_CLIENT_SECRET") as string,
+    userAgent = getEnvironmentVariable("REDDIT_USER_AGENT") as string,
+    searchQueries,
+    mode,
+    categories = ["new"],
+    numberPosts = 10,
+  }: RedditAPIConfig & {
+    searchQueries: string[];
+    mode: string;
```

Comment:
Would be a lot nicer if these were typed instead of just a string since other values with cause errors to be thrown.

```suggestion
    mode: "subreddit" | "username";
```